src/worker/tasks/nonceHealthCheckWorker.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  getLastUsedNonceKeys,
+  getOnchainNonce,
+} from "../../server/routes/admin/nonces";
+import { logger } from "../../utils/logger";
+import { redis } from "../../utils/redis/redis";
+import { NonceHealthCheckQueue } from "../queues/nonceHealthCheckQueue";
+import { logWorkerExceptions } from "../queues/queues";
+
+// Configuration
+
+// Number of consecutive periods to check
+// Checking over multiple periods helps to avoid false positives due to timing issues
+const CHECK_PERIODS = 3;
+
+// Frequency of the worker
+const RUN_FREQUENCY_SECONDS = 60; // Run every minute
+
+// Interfaces
+interface NonceState {
+  timestamp: number;
+  onchainNonce: number;
+  internalNonce: number;
+}
+
+interface WalletHealth {
+  walletAddress: Address;
+  chainId: number;
+  isStuck: boolean;
+  onchainNonce: number;
+  internalNonce: number;
+}
+
+// Initialize the worker
+export const initNonceHealthCheckWorker = () => {
+  NonceHealthCheckQueue.q.add("cron", "", {
+    repeat: { pattern: `*/${RUN_FREQUENCY_SECONDS} * * * * *` },
+    jobId: "nonce-health-check-cron",
+  });
+
+  const _worker = new Worker(NonceHealthCheckQueue.q.name, handler, {
+    connection: redis,
+    concurrency: 1,
+  });
+  logWorkerExceptions(_worker);
+};
+
+// Main handler function
+const handler: Processor<any, void, string> = async (job: Job<string>) => {
+  await checkAllWallets(job);
+};
+
+// Check all wallets
+async function checkAllWallets(job: Job<string>) {
+  const lastUsedNonceKeys = await getLastUsedNonceKeys();
+  const walletHealthPromises = lastUsedNonceKeys.map(async (key) => {
+    const { chainId, walletAddress } = splitLastUsedNonceKey(key);
+    await updateNonceHistory(walletAddress, chainId);
+    const isStuck = await isQueueStuck(walletAddress, chainId);
+    const currentState = await getCurrentNonceState(walletAddress, chainId);
+    return {
+      walletAddress,
+      chainId,
+      isStuck,
+      onchainNonce: currentState.onchainNonce,
+      internalNonce: currentState.internalNonce,
+    };
+  });
+
+  const walletHealthResults = await Promise.all(walletHealthPromises);
+  logWalletHealth(walletHealthResults, job);
+}
+
+// Check if a queue is stuck
+async function isQueueStuck(
+  walletAddress: Address,
+  chainId: number,
+): Promise<boolean> {
+  const historicalStates = await getHistoricalNonceStates(
+    walletAddress,
+    chainId,
+    CHECK_PERIODS,
+  );
+
+  // ensure we have enough data to check
+  if (historicalStates.length < CHECK_PERIODS) return false;
+
+  const oldestOnchainNonce = historicalStates[CHECK_PERIODS - 1].onchainNonce;
+
+  // if for every period, the onchain nonce has not changes, and the internal nonce has strictly increased
+  // then the queue is stuck
+  const isStuckForAllPeriods = historicalStates.every((state, index) => {
+    if (index === historicalStates.length - 1) return true; // Last (oldest) state
+
+    const nextState = historicalStates[index + 1];
```

Comment:
Nit: Since states are ordered new -> old, `nextState` is misleading since it's the actually the previous state.

Suggestion: Either `getHistoricalNonceStates()` should return sorted old -> new (use RPUSH instead of LPUSH) or rename this `prevState`